src/Symfony/Component/Scheduler/Task/AbstractTask.php

```diff
@@ -0,0 +1,225 @@
+        $resolver->setAllowedTypes('execution_ending_date', ['int', 'float', \DateTimeInterface::class, 'null']);
+        $resolver->setAllowedTypes('max_duration', ['int', 'float', 'null']);
+        $resolver->setAllowedTypes('nice', ['int', 'float', 'null']);
+        $resolver->setAllowedTypes('output', 'bool');
+        $resolver->setAllowedTypes('shared', ['bool']);
+        $resolver->setAllowedTypes('priority', 'int');
+        $resolver->setAllowedTypes('queued', ['bool']);
+        $resolver->setAllowedTypes('state', 'string');
+        $resolver->setAllowedTypes('timezone', [\DateTimeZone::class, 'null']);
+        $resolver->setAllowedTypes('tracked', ['bool']);
+        $resolver->setAllowedTypes('tags', ['array', 'null']);
+        $resolver->setAllowedValues('expression', function ($expression): bool {
+            return $this->handleTimeRelativeTasks($expression);
+        });
+        $resolver->setAllowedValues('priority', function (int $priority): bool {
+            return $priority <= 1000 && $priority >= -1000;
+        });
+
+        $resolver->setInfo('arrival_time', 'The time when the task is retrieved in order to execute it');
+        $resolver->setInfo('execution_absolute_deadline', 'An addition of the "execution_start_time" and "execution_relative_deadline" options');
+        $resolver->setInfo('execution_relative_deadline', 'The estimated ending date of the task execution (can be a valid \DatetimeInterface expression or a "strtotime" one)');
+        $resolver->setInfo('execution_start_time', 'The start time of the task execution');
+        $resolver->setInfo('execution_ending_date', 'The date where the execution is finished');
+        $resolver->setInfo('execution_computation_time', 'The CPU time necessary to execute this task');
+
+        if (0 === \count($additionalOptions)) {
+            return $resolver->resolve($options);
+        }
+
+        foreach ($additionalOptions as $additionalOption => $allowedTypes) {
+            $resolver->setDefined($additionalOption);
+            $resolver->setAllowedTypes($additionalOption, $allowedTypes);
+        }
+
+        return $resolver->resolve($options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return \DateTimeInterface|string
+     */
+    public function getExpression()
+    {
+        $expression = $this->options['expression'];
+
+        if ($expression instanceof \DateTimeInterface) {
+            return $expression;
+        }
+
+        if (!\is_bool($date = strtotime($expression))) {
+            return \DateTimeImmutable::createFromFormat('U', $date);
+        }
+
+        return $expression;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOptions(): array
+    {
+        return $this->options;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get(string $key, $default = null)
```

Comment:
what do you think, for main options, writing direct getters? I saw you use the "state" many times

for example

getState() { return $this->get('state', null); }